homeassistant/components/simplefin/config_flow.py

```diff
@@ -0,0 +1,110 @@
+from homeassistant.const import CONF_API_TOKEN
+
+from .const import CONF_ACCESS_URL, DOMAIN, LOGGER
+
+
+async def _validate_or_obtain_access_url(input_string: str) -> str:
+    """Validate the input string as an access URL or a claim token and fetch data using SimpleFin.
+
+    Args:
+        input_string (str): The input_string will either be a URL or a base64 encoded claim_token
+
+    Returns:
+        str: The validated access URL - (or throws an error)
+
+    Raises:
+        SimpleFinInvalidAccountURLError: If the input string is an invalid access URL.
+        SimpleFinPaymentRequiredError
+        SimpleFinAuthError
+        SimpleFinInvalidClaimTokenError: If the input string is an invalid claim token.
+        SimpleFinClaimError: If there's an error in claim token processing.
+
+    """
+
+    if not input_string.startswith("http"):
+        LOGGER.info("[Setup Token] - Claiming Access URL")
+        access_url = await SimpleFin.claim_setup_token(input_string)
+
+    else:
+        LOGGER.info("[Access Url] - 'http' string detected")
+        access_url = input_string
+        LOGGER.info("[Access Url] - validating access url")
+        SimpleFin.decode_access_url(access_url)
+
+    LOGGER.info("[Access Url] - Fetching data")
+    simple_fin = SimpleFin(access_url=access_url)
+    await simple_fin.fetch_data()
+    return access_url
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the initial setup of a SimpleFIN integration."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Prompt user for SimpleFIN API credentials."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                user_input[CONF_API_TOKEN] = await _validate_or_obtain_access_url(
+                    user_input[CONF_API_TOKEN]
+                )
+            except SimpleFinInvalidAccountURLError:
+                errors["base"] = "url_error"
+            except SimpleFinInvalidClaimTokenError:
+                errors["base"] = "invalid_claim_token"
+            except SimpleFinClaimError:
+                errors["base"] = "claim_error"
+            except SimpleFinPaymentRequiredError:
+                errors["base"] = "payment_required"
+            except SimpleFinAuthError:
+                errors["base"] = "auth_error"
+
+            entry = await self.async_set_unique_id(user_input[CONF_API_TOKEN])
+            if entry:
+                self.hass.config_entries.async_update_entry(entry, data=user_input)
+                await self.hass.config_entries.async_reload(entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+
+            self._abort_if_unique_id_configured()
+
+            if not errors:
+                return self.async_create_entry(
+                    title="SimpleFIN",
+                    data={CONF_ACCESS_URL: user_input[CONF_API_TOKEN]},
```

Comment:
I'm really not sure what to call it because they are both different and interchangeable:

- If you pass a `claim_token` to the integration it will use it to obtain an `access_url` it will use to access data
- If you pass an `access_url` it will use it to access data

In the back end we store the `access_url` ... but we allow the user to pass either in